src/effects/builtin/echoeffect.cpp

```diff
@@ -222,17 +222,19 @@ void EchoEffect::processChannel(const ChannelHandle& handle, EchoGroupState* pGr
         if (gs.ping_pong < delay_samples / 2) {
             // Left sample plus a fraction of the right sample, normalized
             // by 1 + fraction.
-            pOutput[i] = (bufferedSampleLeft + bufferedSampleRight * pingpong_frac) /
-                         (1 + pingpong_frac);
+            pOutput[i] = static_cast<CSAMPLE>(
```

Comment:
I think it is sufficient to have float math here. If we make pingpong_frac to CSAMPLE_GAIN it will work.